src/charm.py

```diff
@@ -169,34 +171,38 @@ def _on_get_certificate_action(self, event: ActionEvent) -> None:
 
         event.fail(f"Missing or incomplete certificate data for {hostname}")
 
-    @validate_config_and_tls(defer=False, block_on_tls_not_ready=False)
+    @validate_config_and_tls(defer=False)
     def _on_http_backend_available(self, _: HTTPBackendAvailableEvent) -> None:
         """Handle http_backend_available event for reverseproxy integration."""
         self._reconcile()
 
-    @validate_config_and_tls(defer=False, block_on_tls_not_ready=False)
+    @validate_config_and_tls(defer=False)
     def _on_http_backend_removed(self, _: HTTPBackendRemovedEvent) -> None:
         """Handle data_removed event for reverseproxy integration."""
         self._reconcile()
 
     def _reconcile(self) -> None:
         """Render the haproxy config and restart the service."""
-        ha_information = HAInformation.from_charm(self)
-        self._reconcile_ha(ha_information)
-
         proxy_mode = self._validate_state()
         if proxy_mode == ProxyMode.INVALID:
             # We don't raise any exception/set status here as it should already be handled
             # by the _validate_state method
             return
 
+        if self.certificates._tls_relation_created():  # pylint: disable=protected-access
+            # Reconcile certificates in case the certificates relation is present
+            tls_information = TLSInformation.from_charm(self, self.certificates)
```

Comment:
This is already available in a charm attr. See https://github.com/canonical/haproxy-operator/blob/8e83e8c6cdf4f38cd9e50b800d8621365007ad4d/src/charm.py#L194